wordpress-mobile/WordPress-iOS
# Theme search fixes

Closes #4577; closes #4592

Addresses issues noted with theme search by adopting WPSearchController behaviours.

To test:

- Trigger search in themes and ensure status bar remains white text as search bar reveals (#4577)
- Rotate search controller and ensure header resizes appropriately (noted in #4581)
- Search in Reader > Menu "Enter a URL or tag..." then return to Themes and ensure its prompt does not carry over (#4592)
- Conduct some searches to ensure no behaviour regressions from adopting WPSearchController.

This should work much better than the UISearchController subclassing morass...
Needs review: @kurzee
